### PR TITLE
feat(25188): Add Sentry to the frontend 

### PR DESCRIPTION
**Pull request by vanch3d**
See https://hivemq.kanbanize.com/ctrl_board/57/cards/25188/details/

The PR configure `Sentry` for usage within `Edge`.

The configuration follows the approach used for Akita recently so should be easily maintained. 

### Design 
- As with Akita internal discussion, `Sentry` is considered essential and will not require user consent. If that position changes, the consent banner will have to be extended
- The `Sentry` dashboard has been updated to include the `Edge` project: https://hivemq.sentry.io/projects/edge/?project=4507883299602512

### CODE REVIEW

File: hivemq-edge/src/frontend/src/config/sentry.config.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import * as Sentry from '@sentry/react'
+import { useEffect } from 'react'
+import { createRoutesFromChildren, matchRoutes, useLocation, useNavigationType } from 'react-router-dom'
+
+if (import.meta.env.MODE !== 'development')
+  Sentry.init({
+    dsn: 'https://6878b30beacaee0a926f189192dc3e7c@o4507702360670208.ingest.de.sentry.io/4507883299602512',
```
[the remark this answers is not in the document]
Comment (vanch3d) [reply hivemq-edge/src/frontend/src/config/sentry.config.ts:7]: It is ONLY for `production` (i.e. MODE different from `development`)

[the remark this answers is not in the document]
Comment (sfrehse) [reply hivemq-edge/src/frontend/src/config/sentry.config.ts:7]: Yes. When do we put the heap key into the frontend during compilation?

File: hivemq-edge/src/frontend/.env [hunk 2]
```diff
@@ -6,6 +10,6 @@ VITE_FLAG_MOCK_SERVER=false
 VITE_FLAG_FACET_SEARCH=true
 VITE_FLAG_WORKSPACE_FLOW_PANEL=true
 VITE_FLAG_TOPIC_EDITOR_SHOW_BRANCHES=false
-
+s
```
Comment (oli-hivemq) [inline hivemq-edge/src/frontend/.env:13]: Seems like a mistake

File: hivemq-edge/src/frontend/src/config/sentry.config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfrehse) [reply hivemq-edge/src/frontend/src/config/sentry.config.ts:7]: When we have already put the HEAP key into the `.env`-file, why not the sentry key?

**Review by sfrehse — APPROVED** [review]

**Pull request merged**